### PR TITLE
Add zlib-ng 2.0.7

### PR DESCRIPTION
**Pull request by Vertexwahn**
Add zlib-ng 2.0.7.

Resolves: #567

### CODE REVIEW

**Comment by Vertexwahn** [comment]
Need `skip-url-stability-check` label.

File: modules/zlib-ng/2.0.7/patches/add_build_file.patch [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
++        "zutil.c",
++        "zutil.h",
++        "zutil_p.h",
++    ],
++    hdrs = [
++        "zlib.h",
++        ":zconf_gen",
++    ],
++    copts = [
++        #"-std=c11",
```
Comment (fmeum) [inline modules/zlib-ng/2.0.7/patches/add_build_file.patch:76]: Debug leftover?

File: modules/zlib-ng/2.0.7/patches/add_build_file.patch [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
++        "-DZLIB_COMPAT",
++        "-DWITH_GZFILEOP",
++        "-DWITH_OPTIM",
++        "-DWITH_NEW_STRATEGIES",
++        # For local builds you might want to add "-DWITH_NATIVE_INSTRUCTIONS"
++        # here to improve performance. Native instructions aren't enabled in
++        # the default config for reproducibility.
++    ],
++    defines = [
```
Comment (fmeum) [inline modules/zlib-ng/2.0.7/patches/add_build_file.patch:85]: Can this be `local_defines`?

File: modules/zlib-ng/2.0.7/patches/add_build_file.patch [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
++        "uncompr.c",
++        "zbuild.h",
++        "zendian.h",
++        "zutil.c",
++        "zutil.h",
++        "zutil_p.h",
++    ],
++    hdrs = [
++        "zlib.h",
++        ":zconf_gen",
```
Comment (fmeum) [inline modules/zlib-ng/2.0.7/patches/add_build_file.patch:73]: Could this be moved to `srcs` or should downstream targets have access to it?

File: modules/zlib-ng/2.0.7/patches/add_build_file.patch [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
++# https://github.com/llvm/llvm-project/blob/e8af89ef3de45b3058041388267d728c4ade16e8/utils/bazel/third_party_build/zlib-ng.BUILD#L1
++# some modification applied (also licensed under the same license as the original work)
++
++package(
++    default_visibility = ["//visibility:public"],
++)
++
++genrule(
++    # The input template is identical to the CMake output.
++    name = "zconf_gen",
```
Comment (fmeum) [inline modules/zlib-ng/2.0.7/patches/add_build_file.patch:18]: This looks like it could be private.

File: modules/zlib-ng/2.0.7/MODULE.bazel [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module(
+    name = "zlib-ng",
+    version = "2.0.7",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "rules_cc", version = "0.0.9")
```
Comment (fmeum) [inline modules/zlib-ng/2.0.7/MODULE.bazel:7]: Is it possible that this is unused?

**Review by fmeum — APPROVED** [review]

**Pull request merged**